Services/ODataServices.cs

```diff
@@ -0,0 +1,129 @@
+
+        /// <summary>
+        /// Publish an existing content item.
+        /// </summary>
+        /// <param name="id">The item's ID</param>
+        /// <returns></returns>
+        public IRestResponse Publish(Guid id)
+        {
+            var requestUrl = Settings.BaseUrl + EndpointUrl + "(" + id + ")" + "/operation";
+            var body = "{\"action\": \"Publish\",\"actionParameters\": {}}";
+
+            return this.ExecuteSitefinityRequest(Method.POST, requestUrl, body);
+        }
+
+        /// <summary>
+        /// Modify existing item.
+        /// </summary>
+        /// <param name="item">The item.</param>
+        /// <returns>The response of the request.</returns>
+        public IRestResponse Modify(TContent model)
+        {
+            var requestUrl = Settings.BaseUrl + EndpointUrl + "(" + model.ID + ")";
+            var body = JsonConvert.SerializeObject(model, new JsonSerializerSettings { ContractResolver = new ShouldSerializeContractResolver() });
+            return this.ExecuteSitefinityRequest(Method.PATCH, requestUrl, body);
+        }
+
+        /// <summary>
+        /// Delete existing item.
+        /// </summary>
+        /// <param name="item">The item.</param>
+        /// <returns>Returns the response of the request.</returns>
+        public IRestResponse Delete(Guid id)
+        {
+            var requestUrl = Settings.BaseUrl + EndpointUrl + "(" + id + ")";
+
+            return this.ExecuteSitefinityRequest(Method.DELETE, requestUrl, null);
+        }
+
+        /// <summary>
+        /// Gets an existing item.
+        /// </summary>
+        /// <param name="item">The item.</param>
+        /// <returns>Returns the response of the request.</returns>
+        public IRestResponse GetItem(Guid id)
+        {
+            var requestUrl = Settings.BaseUrl + EndpointUrl + "(" + id + ")";
+
+            return this.ExecuteSitefinityRequest(Method.GET, requestUrl, null);
+        }
+
+        /// <summary>
+        /// Performs a get request against the root endpoint of an item.
+        /// </summary>
+        /// <example>
+        /// Get Metadata
+        /// </example>
+        /// <param name="item">The item.</param>
+        /// <returns>The response of the request.</returns>
+        public IRestResponse Get()
+        {
+            var requestUrl = Settings.BaseUrl + EndpointUrl;
+
+            return this.ExecuteSitefinityRequest(Method.GET, requestUrl, null);
+        }
+
+        /// <summary>
+        /// Executes rest request with the default headers required for Sitefinity.
+        /// </summary>
+        /// <param name="method">The Method of the request.</param>
+        /// <param name="url">The url.</param>
+        /// <param name="body">The body.</param>
+        /// <returns>The response of the executed request.</returns>
+        private IRestResponse ExecuteSitefinityRequest(Method method, string url, object body)
+        {
+            var client = new RestClient(url);
```

Comment:
I think this abstraction could build on the request builder concept I mentioned earlier in this review.